tests/examples/test_quantization_24_sparse_w4a16.py

```diff
@@ -0,0 +1,61 @@
+import shlex
+import shutil
+import sys
+from pathlib import Path
+
+import pytest
+
+from tests.examples.utils import (
+    ReadMe,
+    copy_and_run_command,
+    gen_cmd_fail_message,
+    requires_gpu,
+)
+from tests.testing_utils import run_cli_command
+
+
+@pytest.fixture
+def example_dir() -> str:
+    return "examples/quantization_24_sparse_w4a16"
+
+
+@pytest.mark.example
+@requires_gpu
+class TestQuantization24SparseW4A16:
+    """
+    Tests for examples in the "quantization_24_sparse_w4a16" example folder.
+    """
+
+    def test_doc_example_command(self, example_dir: str, tmp_path: Path):
+        """
+        Test for the example command in the README.
+        """
+        readme_path = Path.cwd() / example_dir / "README.md"
+        readme = ReadMe(readme_path)
+
+        command = readme.get_code_block_content(position=2, lang="shell")
+        assert command.startswith("python")
+
+        command = shlex.split(command)
+        result = copy_and_run_command(tmp_path, example_dir, command)
+
+        assert result.returncode == 0, gen_cmd_fail_message(command, result)
+
+    def test_alternative_recipe(self, example_dir: str, tmp_path: Path):
+        """
+        Test for the example command in the README with the alternative recipe file.
+        """
+        shutil.copytree(str(Path.cwd() / example_dir), tmp_path / example_dir)
+
+        # replace recipe file with alternative
+        script_path = tmp_path / example_dir / "llama7b_sparse_w4a16.py"
```

Comment:
nit: it is likely better to consolidate all the raw strings into variables. some of them are repeated and this also guards against typos.

Reply:
So, this is one of very few cases where raw strings aren't variables, and only one of them is re-used, meaning that for two of the three raw strings, swapping to variables would not introduce any safety, only noise. I will update the one string that gets repeated to use a variable.